### PR TITLE
Updates to Documentation

### PR DESCRIPTION
**Pull request by bradjc**
Adds and re-organizes the readmes.

The big change is moving the getting started and setup instructions to their own file. This lets us display information for various types of users.

Also, what's cool about this PR is that you can test it by just viewing the branch on github.

### CODE REVIEW

File: doc/Mutable_References.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 a heap (no dynamic allocation). Tock uses memory containers
 such as the TakeCell abstraction to allow simple code to keep
 the safety properties Rust provides.
 
-## Brief Overview of Borrowing in Rust 
-Ownership and Borrowing are two design features in Rust which 
+## Brief Overview of Borrowing in Rust
+Ownership and Borrowing are two design features in Rust which
 prevent race conditions and make it impossible to write code that produces
 dangling pointers.
```
Comment (ppannuto) [inline doc/Mutable_References.md]: just below this (not in your changes, but I noticed)

"borrows makes" -> "borrows make"

Comment (ppannuto) [inline doc/Mutable_References.md]: also, "by passing pointer[s]" missing s

Comment (bradjc) [inline doc/Mutable_References.md]: Fixed

**Comment by alevy** [comment]
@bradjc is this merge-worthy or are you still adding commits?

**Comment by bradjc** [comment]
Ready to go!

**Pull request merged**